presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -5,35 +5,546 @@ Tuning Presto
-
-* ``node-scheduler.network-topology``:
-  Sets the network topology to use when scheduling splits. "legacy" will ignore
-  the topology when scheduling splits. "flat" will try to schedule splits on the same
-  host as the data is located by reserving 50% of the work queue for local splits.
+When setting up a cluster for a specific workload it may be necessary to adjust the
+following properties to ensure optimal performance:
+
+  * :ref:`distributed-joins-enabled <tuning-pref-general>`
+  * :ref:`query.max-memory <tuning-pref-query>`
+  * :ref:`query.max-memory-per-node <tuning-pref-query>`
+  * :ref:`query.initial-hash-partitions <tuning-pref-query>`
+  * :ref:`task.concurrency <tuning-pref-task>`
+
+.. _tuning-pref-general:
+
+General properties
+------------------
+
+
+``distributed-index-joins-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:** Serves as default value for ``distributed_index_join`` session property. Enabling this property forces to repartition tables used for index join. This causes drop in processing time, but allows to perform much larger joins. Depending of side of index selected for join (preferably right side) only the other must fit into each node memory (after filtering).
+
+
+``distributed-joins-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``true``
+ * **Description:** Use hash distributed joins instead of broadcast joins. Distributed joins require redistributing both tables using a hash of the join key. This can be slower (sometimes substantially) than broadcast joins but allows much larger joins. Broadcast joins require that the tables on the right side of the join after filtering fit in memory on each machine whereas distributed joins only need to fit in distributed memory across all machines. This can also be specified on a per-query basis using the ``distributed_join`` session property.
+
+
+``redistribute-writes``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``true``
+ * **Description:** Force parallel distributed writes. Serves as default value for ``redistribute_writes`` session property. Setting this property will cause write operator to be distributed between nodes. This allows to utilize distributed storage backend especially in case of small number of huge queries.
+
+
+``resources.reserved-system-memory``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``JVM max memory`` * ``0.4``
+ * **Description:** Amount of memory set up as allocation limit for single presto node. Reaching this limit will cause the server to drop operations. Higher value may increase server stability but may cause problems if physical server is used for other purposes as well. In some configurations of host it may even cause presto to be shut down by host if too much memory will be allocated.
+
```

Comment:
Maximum amount of memory available to each Presto node. Reaching this limit will cause the server to drop operations. Higher value may increase Presto's stability, but may cause problems if physical server is used for other purposes. If too much memory is allocated to Presto, the operating system may terminate the process.